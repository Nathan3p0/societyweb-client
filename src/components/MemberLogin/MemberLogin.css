.memberLogin__form h3 {
    color: #fff;
    margin: 2rem 0 1rem 0;
    font-size: 2rem;
    text-align: center;
}


.memberLogin__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.memberLogin__form > li > label {
    flex: 1 0 100px;
    color: #fff;
    font-size: 1.2rem;
}

.memberLogin__form > li > label + * {
    flex: 1 0 220px;
}

.button__container {
    justify-content: flex-end;
}

.memberLogin__form li button:first-of-type {
    margin-right: .5rem;
}

.memberLogin__form li button {
    padding: 12px 24px;
    border: none;
    background: #4ab3dd;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.memberLogin__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.memberLogin__form li input {
    padding: 15px;
    border: 1px solid #00000025;
    border-radius: 5px;
}

.memberLogin__form a {
    color: #fff;
}

.memberLogin__form a:hover {
    color: #ddd;
}

.member__join {
    margin-top: .5rem;
    text-decoration: underline;
    color: #fff;
}

.member__join:hover {
    color: #ccc;
    cursor: pointer;
}