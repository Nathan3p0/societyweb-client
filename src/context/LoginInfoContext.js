import React from 'react';

const LoginInfoContext = React.createContext();

export default LoginInfoContext;
