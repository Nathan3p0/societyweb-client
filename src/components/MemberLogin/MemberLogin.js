import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import AuthApiService from '../../services/auth-api-service';
import LoginInfoContext from '../../context/LoginInfoContext';
import './MemberLogin.css'

class MemberLogin extends Component {
    static defaultProps = {
        memberLoginSuccess: () => { },
        redirect: false
    }

    static contextType = LoginInfoContext;

    state = {
        error: null,
    }

    handleMemberLoginSubmit = (e) => {
        e.preventDefault();
        this.setState({
            error: null
        });

        const { username, password } = e.target;

        AuthApiService.postLogin({
            username: username.value,
            password: password.value
        })
            .then(res => {
                username.value = '';
                password.value = '';
                this.setState({
                    redirect: true
                })
            })
            .catch(res => {
                this.setState({
                    error: res.error
                })
            })
    }

    render() {

        if (this.state.redirect) {
            return <Redirect to='/member' />
        }

        return (
            <form onSubmit={this.handleMemberLoginSubmit}>
                <ul className="memberLogin__form">
                    <h3>Team Member Login</h3>
                    <li>
                        <label htmlFor="username">Username:</label>
                        <input type="text" name="username" id="username" placeholder="username" required />
                    </li>
                    <li>
                        <label htmlFor="password">Password:</label>
                        <input type="password" name="password" id="password" placeholder="password" required />
                    </li>
                    <li>
                        <button type="submit">Login</button>
                    </li>
                    <li>
                        <a href="#" onClick={this.context.memberJoinToggle}>Not a member? Click here to join.</a>
                    </li>
                </ul>
            </form>
        );
    }
}

export default MemberLogin;