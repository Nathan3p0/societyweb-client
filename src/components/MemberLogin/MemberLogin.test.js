import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MemberLogin from './MemberLogin'
import LoginInfoContext from '../../context/LoginInfoContext';

describe('Member Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <LoginInfoContext.Provider value={{ memberJoinToggle: () => { } }}>
                <MemberLogin />
            </LoginInfoContext.Provider>
            , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(
            <LoginInfoContext.Provider value={{ memberJoinToggle: () => { } }}>
                <MemberLogin />
            </LoginInfoContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
})