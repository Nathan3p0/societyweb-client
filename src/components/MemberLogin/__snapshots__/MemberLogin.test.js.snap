// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Login Component Rendering Tests renders to UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <ul
    className="memberLogin__form"
  >
    <h3>
      Team Member Login
    </h3>
    <li>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        id="username"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </li>
    <li>
      <button
        type="submit"
      >
        Login
      </button>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Not a member? Click here to join.
      </a>
    </li>
  </ul>
</form>
`;
